### PR TITLE
[Backport] Backport the RISC-V backend for Dragonwell11

### PR DESCRIPTION
**Pull request by zhengxiaolinX**
Summary: A backport of the RISC-V backend to the master branch.

Test Plan: Hotspot & JDK tier1\~4

Reviewed-by: kuaiwei

Issue: #209

### CODE REVIEW

**Review by kuaiwei — APPROVED** [review]

**Review by kuaiwei — APPROVED** [review]

**Review by kuaiwei — COMMENTED** [review]
review in progress

File: src/hotspot/share/c1/c1_LIR.hpp [hunk 1]
```diff
@@ -1703,6 +1667,57 @@ class LIR_Op2: public LIR_Op {
   virtual void print_instr(outputStream* out) const PRODUCT_RETURN;
 };
 
+class LIR_OpBranch: public LIR_Op2 {
+ friend class LIR_OpVisitState;
+
+ private:
+#ifndef RISCV
+  LIR_Condition _cond;
```
Comment (kuaiwei) [inline src/hotspot/share/c1/c1_LIR.hpp:1675]: _cond is defined in LIR_Op2， it looks no use for all platform

File: make/autoconf/jdk-options.m4 [hunk 2]
```diff
@@ -39,6 +39,8 @@ AC_DEFUN_ONCE([JDKOPT_SETUP_JDK_VARIANT],
 
   if test "x$with_jdk_variant" = xnormal || test "x$with_jdk_variant" = x; then
     JDK_VARIANT="normal"
+  elif test "x$with_jdk_variant" = xthead && test "x$OPENJDK_TARGET_CPU" = xriscv64 ; then
+    JDK_VARIANT="thead"
```
[the remark this answers is not in the document]
Comment (zhengxiaolinX) [reply make/autoconf/jdk-options.m4]: Dropped.

**Review by kuaiwei — APPROVED** [review]

**Comment by zhengxiaolinX** [comment]
GHA issues are not related.

**Pull request merged**